internal/store/builder.go

```diff
@@ -218,150 +223,169 @@ func availableResources() []string {
 }
 
-func (b *Builder) buildResourceQuotaStore() cache.Store {
+func (b *Builder) buildResourceQuotaStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(resourceQuotaMetricFamilies, &v1.ResourceQuota{}, createResourceQuotaListWatch)
 }
 
-func (b *Builder) buildSecretStore() cache.Store {
+func (b *Builder) buildSecretStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(secretMetricFamilies(b.allowLabelsList["secrets"]), &v1.Secret{}, createSecretListWatch)
 }
 
-func (b *Builder) buildServiceStore() cache.Store {
+func (b *Builder) buildServiceStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(serviceMetricFamilies(b.allowLabelsList["services"]), &v1.Service{}, createServiceListWatch)
 }
 
-func (b *Builder) buildStatefulSetStore() cache.Store {
+func (b *Builder) buildStatefulSetStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(statefulSetMetricFamilies(b.allowLabelsList["statefulsets"]), &appsv1.StatefulSet{}, createStatefulSetListWatch)
 }
 
-func (b *Builder) buildStorageClassStore() cache.Store {
+func (b *Builder) buildStorageClassStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(storageClassMetricFamilies(b.allowLabelsList["storageclasses"]), &storagev1.StorageClass{}, createStorageClassListWatch)
 }
 
-func (b *Builder) buildPodStore() cache.Store {
+func (b *Builder) buildPodStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(podMetricFamilies(b.allowLabelsList["pods"]), &v1.Pod{}, createPodListWatch)
 }
 
-func (b *Builder) buildCsrStore() cache.Store {
+func (b *Builder) buildCsrStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(csrMetricFamilies(b.allowLabelsList["certificatesigningrequests"]), &certv1.CertificateSigningRequest{}, createCSRListWatch)
 }
 
-func (b *Builder) buildValidatingWebhookConfigurationStore() cache.Store {
+func (b *Builder) buildValidatingWebhookConfigurationStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(validatingWebhookConfigurationMetricFamilies, &admissionregistrationv1.ValidatingWebhookConfiguration{}, createValidatingWebhookConfigurationListWatch)
 }
 
-func (b *Builder) buildVolumeAttachmentStore() cache.Store {
+func (b *Builder) buildVolumeAttachmentStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(volumeAttachmentMetricFamilies, &storagev1.VolumeAttachment{}, createVolumeAttachmentListWatch)
 }
 
-func (b *Builder) buildVPAStore() cache.Store {
+func (b *Builder) buildVPAStore() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(vpaMetricFamilies(b.allowLabelsList["verticalpodautoscalers"]), &vpaautoscaling.VerticalPodAutoscaler{}, createVPAListWatchFunc(b.vpaClient))
 }
 
-func (b *Builder) buildLeases() cache.Store {
+func (b *Builder) buildLeases() []*metricsstore.MetricsStore {
 	return b.buildStoreFunc(leaseMetricFamilies, &coordinationv1.Lease{}, createLeaseListWatch)
 }
 
 func (b *Builder) buildStore(
```

Comment:
We might want to rename this function to buildStores

Reply:
I renamed it to `buildStores` in a different commit since the change is a bit bigger.